### PR TITLE
config: set bytes_per_sync & wal_bytes_per_sync

### PR DESCRIPTION
**Pull request by UncP**
fix issue https://github.com/pingcap/tikv/issues/2553#event-1372612179

### CODE REVIEW

File: etc/config-template.toml [hunk 1]
```diff
@@ -245,8 +245,12 @@
 # set backup path, if not set, use "backup" under store path.
 # backup-dir = "/tmp/tikv/store/backup"
 
+# Allows OS to incrementally sync files to disk while they are being
+# written, asynchronously, in the background.
+# bytes-per-sync = "1MB"
+
 # Allows OS to incrementally sync WAL to disk while it is being written.
-# wal-bytes-per-sync = 0
+# wal-bytes-per-sync = "32KB"
```
Comment (siddontang) [inline etc/config-template.toml]: 64KB may be better in our benchmark?

File: etc/config-template.toml [hunk 2]
```diff
@@ -245,8 +245,12 @@
 # set backup path, if not set, use "backup" under store path.
 # backup-dir = "/tmp/tikv/store/backup"
 
+# Allows OS to incrementally sync files to disk while they are being
+# written, asynchronously, in the background.
+# bytes-per-sync = "1MB"
```
Comment (siddontang) [inline etc/config-template.toml]: do we bench this?

File: etc/config-template.toml (hunk 1 above)
Comment (UncP) [reply etc/config-template.toml]: I checked the doc, 32KB is better.

File: etc/config-template.toml (hunk 2 above)
Comment (UncP) [reply etc/config-template.toml]: No, we haven't, but this is a default number in rocksdb if we set rate limiter, so I think 1MB is fine.

**Comment by siddontang** [comment]
PTAL @huachaohuang

**Comment by huachaohuang** [comment]
The code looks good to me, but it changes the default behavior, better to have a performance regression test.

**Comment by UncP** [comment]
I can run some benchmark on `bytes-per-sync` on rocksdb to see whether `1MB` is a good choice.

File: etc/config-template.toml (hunk 1 above)
Comment (zhangjinpeng1987) [reply etc/config-template.toml]: Depends on my benching result in raftengine, 512KB can achieve good performance, also can reduce IO operations per minutes. 512KB is a recommended value, but you also need benching result base on rocksdb. When you are benching, please don't just focus on QPS, you also need to pay close attentions to IOPS and utilization of disk.

**Comment by zhangjinpeng1987** [comment]
As I have tested, `wal_bytes_per_sync` should be various in different type of disks. In traditional mechanical disk, `wal_bytes_per_sync` should be set larger than SSD, or mechanical disk's IOPS will not satisfy the high write load, and the write performance will drop sharply.
So I suggest you to bench `wal_bytes_per_sync` comprehensively.

**Comment by UncP** [comment]
@zhangjinpeng1987 @huachaohuang @siddontang PTAL

**Comment by huachaohuang** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**